src/main/java/org/opensearch/timeseries/rest/handler/AbstractTimeSeriesActionHandler.java

```diff
@@ -398,22 +407,215 @@ protected void validateTimeField(boolean indexingDryRun, ActionListener<T> liste
+            handlePutRequest(indexingDryRun, listener);
         } else {
-            createConfig(indexingDryRun, listener);
+            handlePostRequest(indexingDryRun, listener);
+        }
+    }
+
+    private void handlePutRequest(boolean indexingDryRun, ActionListener<T> listener) {
+        handler
+            .confirmJobRunning(
+                clusterService,
+                client,
+                id,
+                listener,
+                () -> { updateConfig(id, indexingDryRun, listener); },
+                xContentRegistry
+            );
+    }
+
+    private void handlePostRequest(boolean indexingDryRun, ActionListener<T> listener) {
+        createConfig(indexingDryRun, ActionListener.wrap(createConfigResponse -> {
+            if (shouldHandleFlattening(indexingDryRun)) {
+                String configId = RestHandlerUtils.getConfigIdFromIndexResponse(createConfigResponse);
+                String flattenedResultIndexAlias = timeSeriesIndices
+                    .getFlattenedResultIndexAlias(config.getCustomResultIndexOrAlias(), configId);
+                String pipelineId = timeSeriesIndices.getFlattenResultIndexIngestPipelineId(configId);
+
+                timeSeriesIndices
+                    .initFlattenedResultIndex(
+                        flattenedResultIndexAlias,
+                        ActionListener.wrap(initResponse -> setupIngestPipeline(configId, ActionListener.wrap(pipelineResponse -> {
+                            updateResultIndexSetting(
+                                pipelineId,
+                                flattenedResultIndexAlias,
+                                ActionListener.wrap(updateResponse -> listener.onResponse(createConfigResponse), listener::onFailure)
+                            );
+                        }, listener::onFailure)), listener::onFailure)
+                    );
+            } else {
+                listener.onResponse(createConfigResponse);
+            }
+        }, listener::onFailure));
+    }
+
+    private boolean shouldHandleFlattening(boolean indexingDryRun) {
+        Boolean flattenResultIndexMapping = config.getFlattenResultIndexMapping();
+
+        return !indexingDryRun && config.getCustomResultIndexOrAlias() != null && Boolean.TRUE.equals(flattenResultIndexMapping);
+    }
+
+    protected void setupIngestPipeline(String configId, ActionListener<T> listener) {
+        String flattenedResultIndexAlias = timeSeriesIndices.getFlattenedResultIndexAlias(config.getCustomResultIndexOrAlias(), configId);
+        String pipelineId = timeSeriesIndices.getFlattenResultIndexIngestPipelineId(configId);
+
+        try {
+            BytesReference pipelineSource = createPipelineDefinition(flattenedResultIndexAlias);
+
+            PutPipelineRequest putPipelineRequest = new PutPipelineRequest(pipelineId, pipelineSource, XContentType.JSON);
+
+            client.admin().cluster().putPipeline(putPipelineRequest, ActionListener.wrap(response -> {
```

Comment:
I think use client.admin() is fine as I cannot find other ways of put pipeline with client and core's put pipeline api used the same thing: https://github.com/opensearch-project/OpenSearch/blob/32a88eb16bb7bff15fa5430c1490cd102c6cca28/server/src/main/java/org/opensearch/rest/action/ingest/RestPutPipelineAction.java#L77